src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -171,35 +174,45 @@ public LectureResource(LectureRepository lectureRepository, LectureService lectu
         if (!authCheckService.isAtLeastStudentInCourse(course, user)) {
             return forbidden();
         }
+        lecture = filterLectureContentForUser(lecture, user);
+
+        return ResponseEntity.ok(lecture);
+    }
+
+    private Lecture filterLectureContentForUser(Lecture lecture, User user) {
         lecture = lectureService.filterActiveAttachments(lecture, user);
 
-        // to make sure that the information provided for lecture units is equal to the one in the course dashboard
-        // ToDo Improve performance by constructing a more precise call
-        course = this.courseResource.getCourseForDashboard(course.getId());
-        Set<Exercise> exercisesUserIsAllowedToSee = course.getExercises();
+        Set<Exercise> relatedExercises = lecture.getLectureUnits().stream().filter(Objects::nonNull).filter(lectureUnit -> lectureUnit instanceof ExerciseUnit)
```

Comment:
```suggestion
        Set<Exercise> relatedExercises = lecture.getLectureUnits().stream().filter(lectureUnit -> lectureUnit instanceof ExerciseUnit)
```

Reply:
I know but the extra Objects::null does not hurt if somebody else ever changes this code. I added an explanation in 3e36d21f6ca4cb891fc83222af240770dbfcbc69